### DISCUSSION TITLE
GPG signature lost when merging a PR

### DISCUSSION THREAD

**Discussion by Primajin** [community/community#10410] [Pull Requests]
I have all my commits GPG signed.
It shows correctly in the commits *of the PR*
![pr](https://user-images.githubusercontent.com/1742115/150565842-08d57b91-22ec-4a84-89c4-3259c293eedc.png)
Once that PR is merged into `main` it loses all GPG signatures.
![main](https://user-images.githubusercontent.com/1742115/150565862-b7cabf34-360b-4651-8f48-0200c3084c14.png)
Since I have vigilant mode active the PR commits in main are flagged.
I would want the PR to be GPG signed when merged by me.

**Comment by sdavids** [comment]
GitHub should not use the `--no-ff` rebase option, thereby preserving the existing signed commits.

https://git-scm.com/docs/git-rebase#Documentation/git-rebase.txt---force-rebase

All the amazing work you did on [Managing commit signature verification](https://docs.github.com/en/authentication/managing-commit-signature-verification) is practically useless when you use your PR workflow which almost every open source project does.

**Comment by Primajin** [comment]
I had a chat with the support and they came back with this:

> I've had a look into this you can use merging strategies `squash and merge` and `merge commit` to make sure your commits are verified.
>
> I do agree that if these commits through the UI are verified, then `rebase and merge` should be too! This is something our engineering team are aware of and will hopefully be available in the future.

**Comment by castarco** [comment]
Related to this one: https://github.com/github/feedback/discussions/5524
